hnn_core/gui.py

```diff
@@ -0,0 +1,1028 @@
+    if not (plot_type['type'] == 'change' and plot_type['name'] == 'value'):
+        return
+
+    with _plot_out:
+        if plot_type['new'] == 'spikes':
+            fig, ax = plt.subplots()
+            variables['net'].cell_response.plot_spikes_raster(ax=ax)
+
+        elif plot_type['new'] == 'current dipole':
+            fig, ax = plt.subplots()
+            # variables['dpls'][0].plot(ax=ax)
+            plot_dipole(variables['dpls'], ax=ax, average=True)
+
+        elif plot_type['new'] == 'input histogram':
+            # BUG: got error here, need a better way to handle exception
+            fig, ax = plt.subplots()
+            variables['net'].cell_response.plot_spikes_hist(ax=ax)
+
+        elif plot_type['new'] == 'PSD':
+            fig, ax = plt.subplots()
+            variables['dpls'][0].plot_psd(fmin=0, fmax=50, ax=ax)
+
+        elif plot_type['new'] == 'spectogram':
+            freqs = np.arange(10., 100., 1.)
+            n_cycles = freqs / 8.
+            fig, ax = plt.subplots()
+            variables['dpls'][0].plot_tfr_morlet(freqs,
+                                                 n_cycles=n_cycles,
+                                                 ax=ax)
+        elif plot_type['new'] == 'network':
+            fig = plt.figure()
+            ax = fig.add_subplot(111, projection='3d')
+            variables['net'].plot_cells(ax=ax)
+
+
+def load_drives(variables, params, log_out, drives_out, drive_widgets,
+                drive_boxes, tstop):
+    """Add drive ipywidgets from params.
+    """
+    variables['net'] = jones_2009_model(params)
+    log_out.clear_output()
+    with log_out:
+        drive_specs = _extract_drive_specs_from_hnn_params(
+            variables['net']._params, list(variables['net'].cell_types.keys()))
+
+        # clear before adding drives
+        drives_out.clear_output()
+        while len(drive_widgets) > 0:
+            drive_widgets.pop()
+            drive_boxes.pop()
+
+        drive_names = sorted(drive_specs.keys())
+        for idx, drive_name in enumerate(drive_names):  # order matters
+            specs = drive_specs[drive_name]
+            print(f"""load drive:
+                (name={drive_name},
+                type={specs['type']},
+                seed={specs['event_seed']},
+                space_constant={specs['space_constant']}
+                )
+                """)
+            add_drive_widget(
+                specs['type'].capitalize(),
+                drive_boxes,
+                drive_widgets,
+                drives_out,
+                tstop,
+                specs['location'],
+                prespecified_drive_name=drive_name,
+                prespecified_drive_data=specs['dynamics'],
+                prespecified_weights_ampa=specs['weights_ampa'],
+                prespecified_weights_nmda=specs['weights_nmda'],
+                prespecified_delays=specs['synaptic_delays'],
+                render=idx == len(drive_names) - 1,
```

Comment:
this is very confusing to read. Assign `render` above and then use it in the call here.